File: 프로그래머스 풀이/lv2_전화번호_목록.java
import java.util.Arrays;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        
        int i = 1;
        String str = "";
        int len = 0;
        int len_back = 0;
        
        while( i < phone_book.length ) {
            len = phone_book[i-1].length();
            len_back = phone_book[i].length();
            
            if (len > len_back) {
                i++;
                continue;
            }
            str = phone_book[i].substring(0, len);
            
            if (str.equals(phone_book[i-1])) {
                answer = false;
                return answer;
            }
            i++;
        }
        
        return answer;
    }
}

File: 프로그래머스 풀이/lv2_전화번호_목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    int len = 0;
    sort(phone_book.begin(), phone_book.end());
    
    for (int i = 1; i < phone_book.size(); i++) {
        len = phone_book[i-1].length();
        if (phone_book[i-1] == phone_book[i].substr(0, len))
            return false;
    }
    return answer;
}